zuulcilint/__main__.py

```diff
@@ -101,39 +101,70 @@ def get_all_jobs(zuul_yaml_files: list[pathlib.Path]) -> list[list[str]]:
     return all_jobs
 
 
-def print_warnings(bad_yml_files: list[str], duplicated_jobs: set[str]) -> None:
+def print_warnings(bad_yml_files: list[str], duplicated_jobs: set[str], severity="warning") -> None:
     """Print warnings."""
-    nr_warnings = len(bad_yml_files) + len(duplicated_jobs)
-    print(f"Total warnings: {nr_warnings}")
+    n_bad = len(bad_yml_files)
+    n_dup = len(duplicated_jobs)
+
+    if(n_bad + n_dup == 0):
+        return
+
+    if(severity == "warning"):
+        zuul_utils.print_bold("Warnings", severity)
+        print(f"Total {severity}s: {n_bad + n_dup}")
+
     if bad_yml_files:
+        zuul_utils.print_bold(f"File extension {severity}s:", severity)
         zuul_utils.print_bold(
-            f"Found {len(bad_yml_files)} files with 'yml' extension",
+            f"Found {n_bad} files with 'yml' extension",
             None,
         )
         for file_path in bad_yml_files:
             print(f"  {file_path}")
     if duplicated_jobs:
-        zuul_utils.print_bold(f"Found {len(duplicated_jobs)} duplicate jobs", None)
+        zuul_utils.print_bold(f"Duplicate job {severity}s:", severity)
+        zuul_utils.print_bold(f"Found {n_dup} duplicate jobs", None)
         for job in duplicated_jobs:
             print(f"  {job}")
 
 
 def print_results(
-    errors: dict,
-    invalid_playbook_paths: list[str],
+    results: dict,
+    bad_yaml_files,
+    duplicated_jobs,
+    warnings_as_errors,
+    ignore_warnings,
 ) -> None:
     """Print the linting results."""
-    nr_errors = errors["yaml"] + errors["playbook_paths"]
-    print(f"Total errors: {nr_errors}")
-    print(f"YAML errors: {errors['yaml']}")
-    print(f"Playbook path errors: {errors['playbook_paths']}")
-    if invalid_playbook_paths:
-        print("Invalid playbook paths:")
-        for path in invalid_playbook_paths:
-            print(f"  {path}")
+    total_errs = results["errors"]["yaml"] + results["errors"]["playbook_paths"]
+    extra_msg = ""
```

Comment:
```suggestion
    total_yaml_errors = results["errors"]["yaml"]
    total_playbook_path_errors = results["errors"]["playbook_paths"]
    total_warnings = results["warnings"]["file_extension"] + results["warnings"]["duplicated_jobs"]

    extra_msg = ""
```

Store all values to prevent multiple access to dict